### DISCUSSION TITLE
Possible bug in 2.10? (Cannot connect from Guacamole)

### DISCUSSION THREAD

**Discussion by horvathcsabalaszlo** [FreeRDP/FreeRDP#9938] [General]
Hi,

I have been using Apache Guacamole until now to connect to my home compuers. Most of them are running Debian (Bookorm and Trixie), some Windows. The Guacamole server is Debian Bookworm. 
So after upgrading my main computer to Trixie, the RDP connection did not work. Before VNC was used, which worked well. 

I have done a lot of research, and narrowed down the issue to possibly to freerdp2-2.10 client package. 
The symptom is that when the Guacamole tries to connect, it cannot. But only to those, where the OS version is Debian Trixie. Connection to Bookworm or Windows is working via RDP.

The difference between Debian Bookworm and Trixie is the version of gnome-remote-desktop , which is 43 on Bookworm, 44 on Trixie, ad the libfreerdp2 , which is 2.10 on Bookworm, 2.11 on Trixie.

I described the issue here, with logs : 
[https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/issues/183](https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/issues/183)

The main problem is that the remote machine log says 
** (gnome-remote-desktop-daemon:3041831): WARNING **: 23:12:32.514: [RDP] Client did not advertise support for the Graphics Pipeline, closing connection
[23:12:32:248] [3041831:3041930] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
** Message: 23:12:32.514: Unable to check file descriptor, closing connection

But the problem may be in libfreerdp2, because i tried with xrdp,which i can connect to, i got the authorization window, and disconnects immediately. The common point is libfreerdp2. 
Don't need to say, i can connect to my main computer with any RDP client, from Windows and Linux. 
I set up a test Guacamole install, but it also could not connect to my main computer. 

Can you please check what can cause such a problem? If any more information is needed, let me know :)

Thanks in advance :)

**Comment by akallabeth** [comment]
just like @pnowack said, for some reason the gfx pipeline is deactivated.
better ask the people at guacamole about that or check settings, maybe you can activate it somewhere in the config.

if you can reproduce this with our clients (or even remmina) you might have something we can help you with.

**Reply by horvathcsabalaszlo** [reply]
Remmina connects, and all other RDP clients are connecting well. I have asked in Guacamole mailing list, but there was not only no answer, even seems there is no knowledge of this. Guacamole is using libfreerdp2 as a library. 

The only suspicious thing is that all versions of Guacamole (guacd) could not connect to Debian Trixie machines, but can connect to Windows and Debian Bookworm, no matter if these are running xrdp or gnome-remote desktop, so the one common point is left that libfreerdp2 is used by all methods, but the libfreerdp2 is 2.10 on my home gateway server, and 2.11 on clients where it can't connect. Where it can connect, the libfreerdp2 is 2.10.

**Reply by akallabeth** [reply]
wrong,
the common point is guac using ancient graphics paths that are no longer supported by `g-r-d`
windows allows fallback to these (but it is awfully slow)

**Reply by horvathcsabalaszlo** [reply]
Hm, thanks for the information. Then it was misleading with the only common difference.
The bad thing is that i cannot reach guacd developers :D 

I haven't experienced a serious slowness, but when i tested the connection to Bookworm, there were glitches.

**Comment by horvathcsabalaszlo** [comment]
> wrong, the common point is guac using ancient graphics paths that are no longer supported by `g-r-d` windows allows fallback to these (but it is awfully slow)

Sry, just for information. Was there any change in 2.10 - 2.11 regarding graphics pipeline, or fallback options? I think this will be information for Guacd developers.

**Reply by akallabeth** [reply]
wrong question.
`g-r-d` was updated and the older graphics paths ditched due to maintenance overhead and bad user experience with the slow graphics. (@pnowack correct me if I am wrong)

so nothing else changed but now the lack of graphics pipeline is a problem no longer ignorable

**Reply by horvathcsabalaszlo** [reply]
Thanks. The information was that libfreerdp2 is not changed :D 
So originally i was at a good place as g-r-d Git. :D
